src/methods.jl

```diff
@@ -178,12 +178,28 @@ function Dist.mode(d::UnivariateFinite)
     return d.decoder(m)
 end
 
+function Dist.modes(d::UnivariateFinite{S,V,R,P}) where {S,V,R,P}
+    dic = d.prob_given_ref
+    p = values(dic)
+    max_prob = maximum(p)
+    M = R[] # modes
+
+    # see comment in `mode` above
+    throw_nan_error_if_needed(max_prob)
+    for (x, prob) in dic
+        if prob == max_prob
+            push!(M, x)
+        end
+    end
+    return d.decoder(M)
+end
+
 function throw_nan_error_if_needed(x)
     if isnan(x)
         throw(
             DomainError(
                 NaN,
-                "`mode` is invalid for `UnivariateFininite` distribution "*
+                "`mode(s)` is invalid for a `UnivariateFinite` distribution "*
                 "with `pdf` containing `NaN`s"
             )
         )
```

Comment:
For better maintenance/testing going forward, can we please define the exception as a constant, as below, and then improve the corresponding test as in my "Suggestion 1" below?

```julia
      const ERR_NAN_FOUND =  DomainError(
                 NaN,
                 "`mode` is invalid for `UnivariateFininite` distribution "*
                 "`mode(s)` is invalid for a `UnivariateFinite` distribution "*
                 "with `pdf` containing `NaN`s"
)
```

Reply:
So `throw_nan_error` can just `throw(ERR_NAN_FOUND)`.